module/VuFind/tests/unit-tests/src/VuFindTest/Log/Writer/SlackTest.php

```diff
@@ -42,51 +42,41 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
  */
+
 class SlackTest extends \PHPUnit\Framework\TestCase
 {
     /**
      * Test writer functionality
      *
      * @return void
      */
-    public function testWriter(): void
+    public function testHandler(): void
     {
-        // Set up data and expectations:
-        $fakeUri = 'http://fake';
-        $expectedBody = json_encode(
-            [
-                'channel' => '#test',
-                'username' => 'TestName',
-                'text' => ':rotating_light: Formatted message.' . PHP_EOL,
-            ]
-        );
-        $message = ['message' => 'test', 'priority' => 1];
-        $options = ['name' => 'TestName', 'channel' => '#test'];
+        $fakeUri = 'http://fake/webhook';
 
-        // Set up mock client:
-        $client = $this->getMockBuilder(Client::class)
-            ->disableOriginalConstructor()->getMock();
-        $client->expects($this->once())->method('setUri')
-            ->with($this->equalTo($fakeUri));
-        $client->expects($this->once())->method('setMethod')
-            ->with($this->equalTo('POST'));
-        $client->expects($this->once())->method('setEncType')
-            ->with($this->equalTo('application/json'));
-        $client->expects($this->once())->method('setRawBody')
-            ->with($this->equalTo($expectedBody));
-        $client->expects($this->once())->method('send');
+        $logRecord = new LogRecord(
+            datetime: new \DateTimeImmutable('2025-07-09T14:57:30+00:00'),
+            channel: 'test',
+            level: Level::Alert,
+            message: 'test',
+            context: [],
+            extra: []
+        );
+        $handler = $this->getMockBuilder(SlackWebhookHandler::class)
+            ->setConstructorArgs([
+                $fakeUri,
+                '#test',
+                'TestName',
+            ])
+            ->onlyMethods(['write'])
+            ->getMock();
 
-        // Set up mock formatter:
-        $formatter = $this->getMockBuilder(Simple::class)
-            ->disableOriginalConstructor()->getMock();
-        $formatter->expects($this->once())->method('format')
-            ->with($this->equalTo($message))
-            ->will($this->returnValue('Formatted message.'));
+        $handler->expects($this->once())
+            ->method('write')
+            ->with($this->callback(function (LogRecord $record) {
+                return $record->message === 'test';
```

Comment:
Would it make more sense to:

```suggestion
                $this->assertEquals('test', $record->message);
                return true;
```

...? If so, are there other assertions that would be worth making to confirm expected behavior, or is one good enough?